src/pages/Counter/containers/CounterContainer.jsx

```diff
@@ -0,0 +1,56 @@
+import { Component } from "react";
+import CounterView from "../components/CounterView";
+// import Layout from "../components/Layout";
+
+class CounterContainer extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			countValue: 0,
+			isEven: true,
+		};
+	}
+
+	componentDidUpdate() {
+		if (this.state.countValue % 2 === 0 && this.state.isEven !== true) {
+			this.setState({isEven: true});
+		} else if (this.state.countValue % 2 !== 0 && this.state.isEven !== false) {
+			this.setState({isEven: false});
+		}
```

Comment:
simplify this. Here will be enough to compare previous countValue with the current and when this value changes, recalculate parity